packages/eas-cli/src/commands/build/delete.ts

```diff
@@ -0,0 +1,122 @@
+import gql from 'graphql-tag';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { ensureBuildExistsAsync, fetchBuildsAsync, formatBuild } from '../../commandUtils/builds';
+import { ExpoGraphqlClient } from '../../commandUtils/context/contextUtils/createGraphqlClient';
+import { EASNonInteractiveFlag } from '../../commandUtils/flags';
+import { withErrorHandlingAsync } from '../../graphql/client';
+import { Build, DeleteBuildMutation, DeleteBuildMutationVariables } from '../../graphql/generated';
+import Log from '../../log';
+import { ora } from '../../ora';
+import { getDisplayNameForProjectIdAsync } from '../../project/projectUtils';
+import { confirmAsync, selectAsync } from '../../prompts';
+
+async function deleteBuildAsync(
+  graphqlClient: ExpoGraphqlClient,
+  buildId: string
+): Promise<Pick<Build, 'id'>> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .mutation<DeleteBuildMutation, DeleteBuildMutationVariables>(
+        gql`
+          mutation DeleteBuildMutation($buildId: ID!) {
+            build(buildId: $buildId) {
+              deleteBuild(buildId: $buildId) {
+                id
+              }
+            }
+          }
+        `,
+        { buildId }
+      )
+      .toPromise()
+  );
+  return data.build!.deleteBuild;
+}
+
+export async function selectBuildToDeleteAsync(
+  graphqlClient: ExpoGraphqlClient,
+  projectId: string,
+  projectDisplayName: string
+): Promise<string | null> {
+  const spinner = ora().start('Fetching builds…');
+
+  let builds;
+  try {
+    builds = await fetchBuildsAsync(graphqlClient, projectId);
+    spinner.stop();
+  } catch (error) {
+    spinner.fail(
+      `Something went wrong and we couldn't fetch the builds for the project ${projectDisplayName}.`
+    );
+    throw error;
+  }
+  if (builds.length === 0) {
+    Log.warn(`There aren't any builds for the project ${projectDisplayName}.`);
+    return null;
+  } else {
+    const buildId = await selectAsync<string>(
+      'Which build do you want to delete?',
+      builds.map(build => ({
+        title: formatBuild(build),
+        value: build.id,
+      }))
+    );
+
+    return (await confirmAsync({
+      message: 'Are you sure you want to delete it?',
+    }))
+      ? buildId
+      : null;
+  }
+}
+
+export default class BuildDelete extends EasCommand {
+  static override description = 'delete a build';
+  static override args = [{ name: 'BUILD_ID' }];
+  static override flags = {
```

Comment:
Maybe we can allow users to filter by platform and profile as well?

Reply:
WDYM? If they provide platform to delete all builds from a platform? That seems rather... prone to human errors

Reply:
Oh, I meant, when displaying the list of builds. So the people can decide what filters should be applied to this list.

Reply:
Added the filter flags for `platform` and `profile`